app/features-json/setup_json_controller.rb

```diff
@@ -8,5 +8,150 @@ class SetupJSONController < APIController
     get "#{HOME}/configured", authenticate: false do
       return json(Services.onboarding_service.correct_setup?)
     end
+
+    # TODO: we need to protect this API endpoint
+    # one the onboarding user flow, we don't have any accounts yet
+    # however, we need to prohibit sending a POST request to this API endpoint
+    # after CI was already set up
+    post HOME.to_s, authenticate: false do
+      # we set this as default so that we don't need duplicate code for error detection
+      params["bot_account"] ||= {}
+
+      encryption_key = params["encryption_key"]
+      bot_token = params["bot_account"]["token"]
+      bot_password = params["bot_account"]["password"] # TODO: What do we need this for
+      config_repo = params["config_repo"].to_s
+      initial_onboarding_token = params["initial_onboarding_token"]
+
+      missing_parameters = {
+        encryption_key: encryption_key,
+        bot_token: bot_token,
+        bot_password: bot_password,
+        config_repo: config_repo,
+        initial_onboarding_token: initial_onboarding_token
```

Comment:
so can we remove the requirement of this param?

>nakhbari:
I never understood what the initial onboarding token was. Why don't we just use the bot token?
>KrauseFx:
I think you're right, we could do that 👍 Will update the existing code base also to reflect this. Anything we're missing @taquitos

Reply:
So seems like for now we decided to leave it as is, and do the refactor in a separate PR, right?